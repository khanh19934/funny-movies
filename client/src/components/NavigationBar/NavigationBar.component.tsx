import React, {useState} from 'react'
import {pick} from "ramda"
import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
    NavItem,
    NavLink
} from 'reactstrap'
import {NavLink as RRNavLink} from "react-router-dom"
import AuthenticationForm from "../AuthenticationForm/AuthenticationForm.component";

import Routes from "../../modules/Navigator/Navigator.route"
import {clearAllToken} from "../../services/localStorage.service";
import {actions as authActions, connect} from "../../context/appStore"

interface IProps {
    isAuthenticate: boolean
}

const NavigationBar: React.FC<IProps> = props => {
    const [isOpen, setIsOpen] = useState<boolean>(false)

    const toggle = () => setIsOpen(!isOpen)

    const handleLogOut = () => {
        clearAllToken()
        authActions.setAuthenticate(false)
    }

    return (
        <div>
            <Navbar fixed={"top"} color="light" light expand="md">
                <NavbarBrand href="/">Funny Movies</NavbarBrand>
                <NavbarToggler onClick={toggle}/>
                <Collapse isOpen={isOpen} navbar>
                    <Nav className="ml-auto" navbar>
                        <NavItem className="mr-4 authenticate-form">
                            <AuthenticationForm/>
                        </NavItem>
                        {props.isAuthenticate && (
                            <React.Fragment>
                                <NavItem>
                                    <NavLink tag={RRNavLink} exact to={Routes.ShareVideo} activeClassName="active">Share
                                        a
                                        movie</NavLink>
                                </NavItem>
                                <NavItem onClick={handleLogOut}>
                                    <NavLink style={{cursor: 'pointer'}}>
                                        Logout
                                    </NavLink>
                                </NavItem>
                            </React.Fragment>
                        )}
                    </Nav>
                </Collapse>
            </Navbar>
        </div>
    )
}

const enhance = connect(pick(['isAuthenticate']))

export default enhance(NavigationBar)
