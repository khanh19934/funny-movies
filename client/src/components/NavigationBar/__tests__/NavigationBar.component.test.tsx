import "@testing-library/jest-dom/extend-expect";
import React from "react";
import {BrowserRouter} from 'react-router-dom'
import {render, fireEvent} from "@testing-library/react"

import {testSnapshots} from "../../../utils/test.util"
import NavigationBar from "../NavigationBar.component";

const NavigationBarComponent: React.FC<{isAuthenticate: boolean}> = (props) => (
    <BrowserRouter>
        <NavigationBar {...props} />
    </BrowserRouter>
)

describe('Navigation Bar', () => {
    testSnapshots(() => (
        NavigationBarComponent
    ), [
        {
            description: 'NavigationBar',
            props: {
                isAuthenticate: false
            }
        },
        {
            description: 'NavigationBar',
            props: {
                isAuthenticate: true
            }
        }
    ])

    test('should show logout and share movie button when user logged in', () => {
        const {getByText} = render(<NavigationBarComponent isAuthenticate={true}/>)

        expect(getByText(`Logout`)).toBeInTheDocument()
        expect(getByText(`Share a movie`)).toBeInTheDocument()
    })

    test('should show login form', () => {
        const {getByText} = render(<NavigationBarComponent isAuthenticate={false}/>)

        expect(getByText(`Login / Register`)).toBeInTheDocument()
    })

    test('Logout button', () => {
        const {getByText} = render(<NavigationBarComponent isAuthenticate={true}/>)
        const button = getByText('Logout')

        fireEvent.click(button)

        expect(window.localStorage.getItem('accessToken')).toEqual(null)
        expect(window.localStorage.getItem('refreshToken')).toEqual(null)

    })

    test('Collapse Button Nav', () => {
        const {getByLabelText, getByText} = render(<NavigationBarComponent isAuthenticate={true}/>)

        const buttonToggle = getByLabelText('Toggle navigation')
        fireEvent.click(buttonToggle)

        expect(getByText('Logout')).toBeInTheDocument()


    })
})