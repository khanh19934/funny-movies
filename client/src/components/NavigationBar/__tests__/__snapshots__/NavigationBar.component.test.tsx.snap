// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Bar snapshots NavigationBar 1`] = `<DocumentFragment />`;

exports[`Navigation Bar snapshots NavigationBar 2`] = `<DocumentFragment />`;
